### PR TITLE
Add Intel McLaren Island board support document

### PR DESCRIPTION
**Pull request by randylintw**

### CODE REVIEW

File: source/supported-hardware/mtl-mcl.rst [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+Creating Slimbootloader binary for capsule image requires the following steps:
+
+Build |SPN| for |UPX12RPLP|::
+
+  python BuildLoader.py build rplp
+
+Edit the 4-byte platform data in Slimbootloader image by *Hexedit* or equivalent hexdecimal editor.
+
+Go to top of TOP SWAP A at address 0xCFFFF4, edit ``14 01 00 AA`` as below
```
Comment (tsaikevin) [inline source/supported-hardware/mtl-mcl.rst:92]: @randylintw should edit 1f 00 00 AA, fyi

File: source/supported-hardware/mtl-mcl.rst [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+
+.. warning:: As you plan to reprogram the SPI flash, it's a good idea to backup the pre-installed BIOS image first.
+
+
+Boot the board and enter BIOS setup menu to get familiar with the board features and settings.
+
+Debug UART
+^^^^^^^^^^^
+
+For |MTL| platforms, serial port connector location can be found from the above table for each supported target board.
```
Comment (tsaikevin) [inline source/supported-hardware/mtl-mcl.rst:46]: @randylintw pls add the table or description of the serial port location.

File: source/supported-hardware/mtl-mcl.rst [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+A capsule image could ecapsulate a Slimbootloader image to be used in firmware update mechanism. More information is described in :ref:`firmware-update`.
+
+For the process running properly, an IFWI image programmed on board with TOP SWAP size configuration is required.
+
+.. note:: Enabling TOP SWAP size configuration requires additional firmware components and tools.
+
+Creating Slimbootloader binary for capsule image requires the following steps:
+
+Build |SPN| for |UPX12RPLP|::
```
Comment (tsaikevin) [inline source/supported-hardware/mtl-mcl.rst:86]: @randylintw replace UPX12 with MCL

Comment (randylintw) [reply source/supported-hardware/mtl-mcl.rst]: rename it, thanks for your review ~

File: source/supported-hardware/mtl-mcl.rst (hunk 2 above)
Comment (randylintw) [reply source/supported-hardware/mtl-mcl.rst]: ok, document the location.

File: source/supported-hardware/mtl-mcl.rst (hunk 1 above)
Comment (randylintw) [reply source/supported-hardware/mtl-mcl.rst]: remove it :)

File: source/supported-hardware/mtl-mcl.rst [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+
+    <BIOS_IMAGE>     : Input file. Factory BIOS extracted from McLaren Island board.
+    <SBL_IFWI_IMAGE> : Output file. New IFWI image with SBL in BIOS region.
+    -p <value>       : 4-byte platform data for platform ID (e.g. 1F) and debug UART port index (e.g. 00).
+
+.. Note:: StitchLoader.py script works only if Boot Guard in the base image is not enabled, and the silicon is not fused with Boot Guard enabled.
+          If Boot Guard is enabled, please use StitchIfwi.py script instead.
+
+Sample Command:
+python Platform\MeteorlakeBoardPkg\Script\StitchIfwi.py  -b fvme
```
Comment (tsaikevin) [inline source/supported-hardware/mtl-mcl.rst:74]: @randylintw I think you should replace '\' with '/'. all the '\' missing in the html after build.

File: source/supported-hardware/mtl-mcl.rst [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+      -s Outputs\mtl\Stitch_Components.zip
+      -c Platform\MeteorlakeBoardPkg\Script\StitchIfwiConfig_mcl.py
+      -w \Stitchifwi_components_mtl -p mtlp -d 0xAA00001F
+
+See :ref:`stitch-tool` on how to stitch the IFWI image with |SPN|.
```
Comment (tsaikevin) [inline source/supported-hardware/mtl-mcl.rst:79]: @randylintw suggest to move this line to line 69 after the stitchloader.py description.

**Review by tsaikevin — APPROVED** [review]

**Pull request merged**